micronota/bfillings/diamond.py

```diff
@@ -6,414 +6,291 @@
+    OptionParam('--min-score',
+                help=('Minimum bit score to keep an alignment. Setting this option'
+                      'will override the --evalue parameter.')),
+    OptionParam('--query-cover',
+                help='Report only alignments above the given percentage of query cover.'),
+    OptionParam('--salltitles',
+                help='Print full length subject titles in output.'),
+
+    OptionParam('--band', help=''),
+    OptionParam('--index-chunks', '-c',
+                help='The number of chunks for processing the seed index.'),
+
+    OptionParam('--sensitive',
+                help=('Trigger the sensitive alignment mode with a 16x9 seed'
+                      'shape config.')),
+    OptionParam('--tmpdir', '-t',
+                help='Directory to be used for temporary storage.'),
+
+    OptionParam('--db', '-d', help='Path to DIAMOND database file'),
+    OptionParam('--daa', '-a',
+                help='Path to DAA file.'),
+    OptionParam('--query', '-q',
+                help=('Path to query input file in FASTA or FASTQ format '
+                      '(may be gzip compressed).'))]
+
+
+makedb_params = [
+    OptionParam('--threads', '-p', name='cpus'),
+    OptionParam('--in', name='fasta',
+                help='protein reference database file in FASTA format (may be gzip compressed)'),
+    OptionParam('--db', '-d',
+                help='DIAMOND database file.'),
+    OptionParam('--block-size',
+                help='Block size in billions of sequence letters to be processed at a time.')]
+
+
+view_params = [
+    OptionParam('--outfmt', name='fmt', action=check_choice(('tab', 'sam')),
+                help=('Format of output file. (tab = BLAST tabular format;'
+                      'sam = SAM format)')),
+    OptionParam('--daa', '-a',
+                help='Path to DAA file.'),
+    OptionParam('--out', '-o',
+                help='Path to output file.'),
+    OptionParam('--compress', action=check_choice((0, 1)),
+                help='Compression for output file (0=none, 1=gzip).')]
+
+
+def run_blast(query, daa, aligner='blastp', **kwargs):
+    '''Search query sequences against the database.
+
+    Parameters
+    ----------
+    query : str
+        The file path of the query seq
+    daa : str
+        The file path of the output daa file
+    aligner : str
+        The aligner. blastp or blastx
+    kwargs : dict
+        keyword arguments. Command line parameters for diamond blastp
+        or blastx.
+    Returns
+    -------
+    str
+        The file path of the blast result.
+    '''
+    logger = getLogger(__name__)
+
+    blast = Dumpling(['diamond', aligner],
+                     params=Parameters(*blast_params))
+    blast.update(query=query, daa=daa, **kwargs)
+    logger.info('Running {}'.format(blast.command))
+    blast()
+    return blast
+
+
+def run_view(daa, out, fmt='sam', **kwargs):
+    '''
```

Comment:
Add 

Runs the view command to convert between the Diamond format and the Blast format.